### PR TITLE
fix partition table index scan

### PR DESCRIPTION
**Pull request by zhexuany**
This is only temp fix, I am investigating is there any better fix. 

In addition to the real fix, I also improve the explain info and refactor the way to construct ScanPlan.

### CODE REVIEW

File: tikv-client/src/main/java/com/pingcap/tikv/util/RangeSplitter.java [hunk 1]
```diff
@@ -79,6 +83,25 @@ public String getHost() {
...
+      return regionTask.host.equals(this.host)
+          && regionTask.ranges.equals(this.ranges)
+          && regionTask.region.equals(this.region)
+          && regionTask.store.equals(this.store);
+    }
+
+    @Override
+    public int hashCode() {
+      int hash = 7;
+      hash = 31 * hash * this.region.hashCode();
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/util/RangeSplitter.java]: I think you should use the following instead.
```suggestion
      hash = 31 * hash + this.region.hashCode();
```

Comment (zhexuany) [reply tikv-client/src/main/java/com/pingcap/tikv/util/RangeSplitter.java]: I already remove this. It seems we don't need this.

File: tikv-client/src/main/java/com/pingcap/tikv/region/TiRegion.java [hunk 2]
```diff
@@ -183,6 +183,26 @@ public Region getMeta() {
...
+  }
+
+  @Override
+  public int hashCode() {
+    int hash = 7;
+    hash = 31 * hash * meta.hashCode();
+    hash = 31 * hash * peer.hashCode();
+    hash = 31 * hash * commandPri.hashCode();
+    hash = 31 * hash * isolationLevel.hashCode();
+    return hash;
```
Comment (marsishandsome) [inline tikv-client/src/main/java/com/pingcap/tikv/region/TiRegion.java]: `return Objects.hash(meta, peer, isolationLevel, commandPri);`

**Comment by zhexuany** [comment]
Maybe file this "suggest to add test: read from partition table with 1000 partitions" in another PR.

**Pull request merged**